Bowling/Bowling/Score.cs

```diff
@@ -0,0 +1,109 @@
+﻿using NUnit.Framework.Constraints;
+using NUnit.Framework.Internal.Execution;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+
+namespace BowlingApp
+{
+    public class Score
+    {
+        public const int frameNumberTotal = 10;
+        public const int frameNumberWithBonusRoll = 10;
+
+        public List<Frame> CreateScoreCard(string bowlingMarks)
+        {
+            var placeCount = 0;
+            List<Frame> framesCollection = new List<Frame>();
+            FrameNoBonusRollFactory frameNoBonusRoll;
+            FrameWithBonusRollFactory frameWithBonusRoll;
+            var marksLength = bowlingMarks.Length;
+
+            for (var frameNumber = 1; frameNumber <= frameNumberTotal; frameNumber++)
+            {
+                Frame frame = new Frame();
+                
+                if(frameNumber != frameNumberWithBonusRoll)
+                {
+                    frameNoBonusRoll = new FrameNoBonusRollFactory();
+                    frame = frameNoBonusRoll.Create(frameNumber, bowlingMarks, placeCount);
+                    placeCount = frameNoBonusRoll.PlaceCountHolder;
+                }
+                else
+                {
+                    frameWithBonusRoll = new FrameWithBonusRollFactory();
+                    frame = frameWithBonusRoll.Create(bowlingMarks, placeCount);
+                }
+
+                framesCollection.Add(frame);
+            }
+
+            return framesCollection;
+        }
+
+        public int CalculateScoreCardTotals(List<Frame> framesCollection)
+        {
+            List<Frame> updatedFramesCollection = UpdateFrameScoresForStrikesAndSpares(framesCollection);
+            var grandTotal = CalculateFramesGrandTotal(updatedFramesCollection);
+
+            return grandTotal;
+        }
+
+        private List<Frame> UpdateFrameScoresForStrikesAndSpares(List<Frame> framesCollection)
+        {
+            Frame currentFrame;
+            Frame oneFrameAfterCurrentFrame;
+            Frame twoFramesAfterCurrentFrame;
+
+            for (var frameNumber = 0; frameNumber <= frameNumberTotal - 2; frameNumber++)
+            {
+                currentFrame = framesCollection[frameNumber];
+                oneFrameAfterCurrentFrame = framesCollection[frameNumber + 1];
+
+                if ((frameNumberTotal - 2) - frameNumber >= 2)
+                {
+                    twoFramesAfterCurrentFrame = framesCollection[frameNumber + 2];
+                }
+                else
+                {
+                    twoFramesAfterCurrentFrame = framesCollection[frameNumberTotal - 1];
+                }
+
+                switch (currentFrame.frameOutcome)
+                {
+                    case (RollTypes.Strike):
+                        if (oneFrameAfterCurrentFrame.frameOutcome == RollTypes.Strike)
```

Comment:
I would encourage you to try and avoid branching logic inside of switch-statements.

Reply:
You got it!  All gone.